notes.md

```diff
@@ -190,6 +190,57 @@ For (possibly) Laced Data
 | 0x00        | **MUST** | Consecutive laced frames |
 Table: Lace data after lace sizes{#blockLacedData}
 
+
+## Random Access Points
+
+Random Access Points (RAP) are positions where the parser can seek to and start playback without decoding
+of what was before. In Matroska `BlockGroups` and `SimpleBlocks` can be RAPs.
+To seek to these elements it is still necessary to seek to the `Cluster` containing them
+and start playback from the `BlockGroup` or `SimpleBlock` that is a RAP.
+
+Because a Matroska File is usually composed of multiple tracks playing at the same time
+-- video, audio and subtitles -- to seek properly to a RAP, each selected track must be
+taken in account. Usually all audio and subtitle `BlockGroup` or `SimpleBlock` are RAP.
+They are independent of each other and can be played randomly.
+
+Video tracks on the other hand often use references to previous and future frames for better
+coding efficiency. Frames with such reference **MUST** either contain one or more
+`ReferenceBlock` Elements in their `BlockGroup` or **MUST** be marked
+as non-keyframe in a `SimpleBlock`; see (#simpleblock-header-flags).
+
+Frames that are RAP -- i.e. they don't depend on other frames -- **MUST** set the keyframe
+flag if they are in a `SimpleBlock` or their parent `BlockGroup` **MUST NOT** contain
+any `ReferenceBlock`.
+
+For legacy reasons, audio and subtitle track **MAY** use a `SimpleBlock` without the keyframe
+flag set, even though they are RAPs. So for audio and subtitle tracks, the only way to specify
+a frame is not a RAP is to put it in a `BlockGroup` with a `ReferenceBlock`. In all other cases
+they **MUST** be considered as decodable on their own.
```

Comment:
This interacts very, very badly with one of the most popular current audio codecs: TrueHD. TrueHD uses two different frame types: regular frames & major sync frames. Decoding can only start from major sync frames. `mkvmerge` and other software has therefore marked frames containing a TrueHD major sync frame with the "keyframe" flag and others without it.

Forcing muxers to use reference blocks here would be a significant increase in file overhead. TrueHD uses a high number of very small frames (both wrt. the number of samples it contains & the its byte size, often having a duration of less than 1ms with a size of around 100 bytes), and the overwhelming majority are regular sync frames, not major ones, in the range of 1 major sync frame every 125 regular frames.

All of those would have to use BlockGroups with reference blocks, which adds about 4% of container overhead for each TrueHD track.

In addition this would suddenly make all existing TrueHD-in-Matroska-with-Simpleblocks files invalid or at least bad to seek in.

What's the reasoning here? Why would we not keep using key frames for audio tracks correctly as there _are_ audio codecs that actually have such a distinction?

I can understand this with subtitle tracks as there's no subtitle codec I know of that uses such a distinction, but changing the semantics for audio tracks in this way is just plain wrong. I'm really against this change in its current form.

Reply:
> What's the reasoning here?

The reason is given in the first sentence: `For legacy reasons`. There are old files out there where the audio SimpleBlock never had the keyframe flag set. Looking at my old files it seems rare though. A deeper analysis is needed to find out what was writing these files and when it was fixed. It seems to be quite old. If we determine it's rare enough we can just add a mention that it exists and is bogus, the rule being to rely on the SimpleBlock flag for RAP, at least for audio.

There's also the question of lacing that I forgot to cover in #496. The keyframe flag **MUST** apply to all the frames in the lace (which are pretty much always audio). Are TrueHD files applying this rule or only the first frame of a lace has the keyframe flag ?

Reply:
After a look at (some of) my older files, I couldn't find audio tracks with SimpleBlock without the keyframe flag. BlockGroup don't have it so it's not an issue, they are all considered keyframes (I don't know about TrueHD). I'll have to dig deeper to find some samples and what/when they were produced (it may be easier to analyze of mkvmerge and libavformat code as well).

Reply:
Removed the legacy section. I could not find any old file that actually had this behavior.